src/legacy/server/saved_objects/lib/import.ts

```diff
@@ -190,20 +186,22 @@ export async function resolveImportConflicts({
   savedObjectsClient,
   replaceReferences,
 }: ResolveImportConflictsOptions): Promise<ImportResponse> {
-  const errors: CustomError[] = [];
+  let errors: CustomError[] = [];
   const filter = createObjectsFilter(skips, overwrites, replaceReferences);
   const objectsToResolve = await collectSavedObjects(readStream, objectLimit, filter);
 
   // Replace references
-  for (const referenceToReplace of replaceReferences) {
-    for (const savedObject of objectsToResolve) {
-      for (const reference of savedObject.references || []) {
-        if (
-          reference.type === referenceToReplace.type &&
-          reference.id === referenceToReplace.from
-        ) {
-          reference.id = referenceToReplace.to;
-        }
+  const refReplacementsMap = replaceReferences.reduce(
```

Comment:
Because of the way TS types reduce, and the fact that the reduce function isn't returning new data on each iteration, I think it would be cleaner to use something like:

```ts
const refReplacementsMap: Record<string, string> = {};
for (const {type, to, from} of replaceReferences) {
  refReplacementsMap[`${type}:${from}`] = to;
}
```